packages/react/src/hooks/useDidUpdate.ts

```diff
@@ -0,0 +1,24 @@
+import { DependencyList, useEffect, useRef } from 'react';
+
+/**
+ *  useDidUpdate hook
+ *
+ *  dependency 배열의 요소가 업데이트 됐을때 콜백 함수를 실해압니다.
+ *
+ * @param {Function} callback dependency 배열의 요소가 업데이트 될 때 호출 할 콜백 함수
+ * @param {Array} conditions 업데이트를 트리거하는 변수 목록
+ */
+export default function useDidUpdate(callback: () => void, dependencies: DependencyList) {
+  const hasMount = useRef(false);
+
+  if (JSON.stringify(dependencies) === '[]') {
+    throw "두번째 인자로 []을 사용하면 'useDidUpdate'는 동작하지 않습니다.";
```

Comment:
```suggestion
    throw new Error('useDidUpdate 훅의 Dependency Array는 빈 배열일 수 없습니다.');
```

일반적으로 catch문으로 에러를 잡을 때 인자로 암시적으로 `Error` 객체가 넘어오는 경우가 많은데다가, 타입스크립트는 catch문의 인자의 타입을 제대로 추론하지 못 하기 때문에 에러 객체가 아닌 타입을 던지게 되면, catch 문에서 `e.message`등에 접근하다가 터지는 경우도 있어유.

그래서 일관적으로 `Error` 객체를 던져주는 것이 좋을 것 같습니다....!

그리고 메세지도 "두 번째 인자"처럼 현재 코드의 구현에 종속되어 있는 부분이 있고, "동작하지 않는다"라는 것도 굳이 설명해주지 않더라도 에러가 터진 시점에서 이미 알 수 있는 사실이니, 현재 에러가 터진 상황과 원인을 설명하도록 에러 메세지가 변경되면 좋을 것 같습니다! (에러를 고칠 수 있는 방법까지 나오면 더 좋구요)

Reply:
에러문구도 이렇게 다를 수 있다니.. 감사합니다 :)